### ISSUE TITLE
Throw error if someone tries to use an engine which is not compatible with a specific OS

### ISSUE THREAD

**Issue by jondubois** [SocketCluster/socketcluster#179]
E.g. If user tries to use uWS on windows, it should throw an error and advise the user to switch to plain ws.

**Comment by efkan** [comment]
Hello @jondubois ,

I had had the version 2.5.3 and it would work properly on Windows.

Today I uninstalled `socketcluster` and I've been trying to install v4.6.2. 

I get the error below;

```
npm ERR! code EBADPLATFORM

npm ERR! notsup Not compatible with your operating system or architecture: uws@0.6.2
npm ERR! notsup Valid OS:    darwin,linux
npm ERR! notsup Valid Arch:  any
npm ERR! notsup Actual OS:   win32
npm ERR! notsup Actual Arch: x64
```

Also I tried to remove 'uWS' dependencies from `package.json` file of `socketcluster-server` which is in `npm-cache` folder.

Unfortunately I couldn't overcome this issue. 

Is there any chance to install the newest version of `SocketCluster` on Windows?

(If I can install the newest version of `socketcluster` then I try to use plain ws)
